### PR TITLE
Reimplement code schools page

### LINKED ISSUE #170: Reimplement /code_schools page
# Feature
## Why is this feature being added?

https://operationcode.org/code_schools exists and it needs to exist in the new code base as well.

## What should your feature do?
- Implement the code_schools page via `pages/code_schools.js`
- Emulate [this mock](https://projects.invisionapp.com/share/KSO12EFRU7E#/screens/323736695_OC_Coding_Schools_Landing_Page).
- [See live source code.](https://github.com/OperationCode/operationcode_frontend/tree/master/src/scenes/home/codeSchools) Feel free to copy-paste a lot.
- Refactor as required to match our new standards (linting, using components where possible, etc.)
- Implement new card design
- Reference moocs from API ([see here](https://github.com/OperationCode/operationcode_backend/pull/250))

### PR DESCRIPTION
**Pull request by markchernov**
# Description of changes

# Issue Resolved

Fixes #170

### CODE REVIEW

**Comment by kylemh** [comment]
🤖 **This is a bot** 🤖<br /><br />🎉 Deployed Storybook preview! 🎉<br /><br />Click the link at the bottom of this comment to see it live.

Built with commit 554da5d8428a817619b8d1f479d4d3ab8f1d7701

https://deploy-preview-314--operation-code-storybook.netlify.com

File: pages/code_schools.js [hunk 1]
```diff
@@ -1,71 +1,172 @@
...
-    schools: [],
+    allSchools: [],
+    filteredSchools: [],
+    selectedStates: [],
   };
 
   async componentDidMount() {
     const { data } = await getCodeSchoolsPromise();
-    this.setState({ schools: data });
+    this.setState({ allSchools: data });
```
Comment (kylemh) [inline pages/code_schools.js]: you can do this in the same line:
```suggestion
    this.setState({ allSchools: data, filteredSchools: data });
```

File: pages/code_schools.js [hunk 2]
```diff
@@ -1,71 +1,172 @@
...
+    this.setState({ filteredSchools: data });
   }
 
+  filterVaApproved = () => {
+    const { allSchools } = this.state;
+    const vaApproved = allSchools.filter(school => {
+      const vaFound = school.locations.some(location => location.va_accepted === true);
```
Comment (kylemh) [inline pages/code_schools.js]: I think the `.some` is small and clear enough to return as a one-liner
```suggestion
      return school.locations.some(location => location.va_accepted === true);
```

File: pages/code_schools.js [hunk 3]
```diff
@@ -1,71 +1,172 @@
...
+      return vaFound;
+    });
+    this.setState({ filteredSchools: vaApproved });
```
Comment (kylemh) [inline pages/code_schools.js]: You can do the state changes in the same line in all of your class methods. 👍

File: pages/code_schools.js [hunk 4]
```diff
@@ -1,71 +1,172 @@
...
+  };
+
+  filterOnline = () => {
+    const { allSchools } = this.state;
+    const onlineSchools = allSchools.filter(school => school.has_online);
+    this.setState({ filteredSchools: onlineSchools });
+    this.setState({ selectedStates: [] });
+  };
+
+  filterMoocs = () => {
```
Comment (kylemh) [inline pages/code_schools.js]: Can you remove this button and logic? The mock doesn't appear to want to filter by MOOCs

Comment (markchernov) [reply pages/code_schools.js]: For me it looked like that Mooc is just like any other online only school. I personally see no reason to separate them into special section. But I guess we could stick them to the bottom of the page like in the old implementation.

File: pages/code_schools.js (hunk 1 above)
Comment (markchernov) [reply pages/code_schools.js]: Yes, all the styles are still screwed up. I just wanted to get your opinion on general layout.

File: pages/code_schools.js (hunk 4 above)
Comment (kylemh) [reply pages/code_schools.js]: It's just about following the mock that design gave at this point. We'll implement analytics and find out if they're suffering. MOOCs are also very distinctly different than the other coding schools and wont actually be listed with the Code School card.

Comment (markchernov) [reply pages/code_schools.js]: Sounds good, will create special mooc card with logo and link button. Not sure where the mooc description is coming from. There is no such filed on incoming json.

Comment (markchernov) [reply pages/code_schools.js]: I'd like to reuse SuccessStory card for mooc school display. I guess there is another story to redesign it:  https://github.com/OperationCode/front-end/issues/291

Comment (kylemh) [reply pages/code_schools.js]: Indeed! Good catch. Your more than welcome to pause here, and finish out that issue.

**Comment by kylemh** [comment]
The point is hopefully that you wont have to. It will be slightly different than the mock in that it will be:

```
<FlatCard imageSource={moocImageUrl} imageAlt={`${moocName} logo`}>
  {moocParagraph}
</FlatCard>
```

File: pages/styles/code_schools.css [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+.moocCardsWrapper {
+  display: flex;
+  flex-wrap: wrap;
+  justify-content: center;
+  align-items: center;
+}
+
+.moocCard {
```
Comment (kylemh) [inline pages/styles/code_schools.css]: Remove everything below this. You won't need to style anything 🍾

File: pages/code_schools.js [hunk 6]
```diff
@@ -1,71 +1,181 @@
...
+                  <OutboundLink href={mooc.url} analyticsEventLabel={`Link to ${mooc.name}`}>
+                    {mooc.name}
+                  </OutboundLink>
+                }
+              >
+                {' '}
+                This is a description of a Mooc School. This is a description of a Mooc School. This
+                is a description of a Mooc School. This is a description of a Mooc School. This is a
+                description of a Mooc School. This is a description of a Mooc School.
+              </FlatCard>
```
Comment (kylemh) [inline pages/code_schools.js]:
```suggestion
<FlatCard
  key={mooc.name}
  imageSource={this.prepUrl(mooc.name)}
  imageAlt={`${mooc.name} logo`}
>
  <>
    This is a description of a Mooc School. This is a description of a Mooc School. This
    is a description of a Mooc School. This is a description of a Mooc School. This is a
    description of a Mooc School. This is a description of a Mooc School.
    <div className={styles.centered}>
      <OutboundLink href={mooc.url} analyticsEventLabel={`Link to ${mooc.name}`}>
        {mooc.name}
      </OutboundLink>
    </div>
  </>
</FlatCard>      
```

File: pages/code_schools.js [hunk 7]
```diff
@@ -160,19 +160,19 @@ export default class CodeSchools extends React.Component {
...
               >
-                {' '}
-                This is a description of a Mooc School. This is a description of a Mooc School. This
-                is a description of a Mooc School. This is a description of a Mooc School. This is a
-                description of a Mooc School. This is a description of a Mooc School.
+                <>
+                  This is a description of a Mooc School. This is a description of a Mooc School.
+                  This is a description of a Mooc School. This is a description of a Mooc School.
+                  This is a description of a Mooc School. This is a description of a Mooc School.
+                  <div className={styles.centered}>
```
Comment (kylemh) [inline pages/code_schools.js]: sorry I wasn't clear, but you'll want to make a class for this in the stylesheet!

```css
.centered {
  margin: 0 auto;
```

should do it i think

File: pages/code_schools.js [hunk 8]
```diff
@@ -1,71 +1,181 @@
...
+    filteredSchools: [],
+    moocSchools: [],
+    selectedStates: [],
   };
 
   async componentDidMount() {
     const { data } = await getCodeSchoolsPromise();
-    this.setState({ schools: data });
+    // should be school.mooc, using school.has_online for testing
+    const moocs = data.filter(school => school.has_online);
```
[the remark this answers is not in the document]
Comment (markchernov) [reply pages/code_schools.js]: ok, hard-coded some moocs

**Comment by markchernov** [comment]
Thanks @kylemh , can I merge it then?

**Pull request merged**